Tests/CommandsTests/WorkspaceTests.swift

```diff
@@ -127,75 +127,40 @@ final class WorkspaceTests: XCTestCase {
         // \ A: checked out (@v1)
         //   \ AA: checked out (@v1)
         // \ B: missing
-        
         mktmpdir { path in
-            // Create the test repositories, we don't need them to have actual
-            // contents (the manifests are mocked).
-            var repos: [String: RepositorySpecifier] = [:]
-            for name in ["A", "AA"] {
-                let repoPath = path.appending(component: name)
-                try makeDirectories(repoPath)
-                initGitRepo(repoPath, tag: "initial")
-                repos[name] = RepositorySpecifier(url: repoPath.asString)
-            }
-
-            // Create the mock manifests.
-            let rootManifest = Manifest(
-                path: AbsolutePath("/UNUSED"),
-                url: path.asString,
-                package: PackageDescription.Package(
-                    name: "Root",
-                    dependencies: [
-                        .Package(url: repos["A"]!.url, majorVersion: 1),
-                        .Package(url: "//B", majorVersion: 1)
+            let graph = try MockManifestGraph(at: path,
+                rootDeps: [
+                    MockDependency("A", version: v1),
+                    MockDependency("B", version: v1),
+                ],
+                packages: [
+                    MockPackage("A", version: v1, dependencies: [
+                        MockDependency("AA", version: v1)
```

Comment:
Also... it is a little unfortunate we have to maintain this piece and then a separate piece for managing the Git repositories. I wonder if there is a way to merge those things (maybe a custom repo provider?)?